src/plugins/discover/public/context_awareness/profile_providers/common/deprecation_logs/profile.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { isOfAggregateQueryType } from '@kbn/es-query';
+import { getIndexPatternFromESQLQuery } from '@kbn/esql-utils';
+import { DataSourceCategory } from '../../../profiles';
+import { DataSourceType, isDataSourceType } from '../../../../../common/data_sources';
+import { type DataSourceProfileProvider } from '../../../profiles';
+import { DEPRECATION_LOGS_PATTERN_PREFIX, DEPRECATION_LOGS_PROFILE_ID } from './consts';
+
+export const createDeprecationLogsDocumentProfileProvider = (): DataSourceProfileProvider<{
+  formatRecord: (flattenedRecord: Record<string, unknown>) => string;
+}> => ({
+  profileId: DEPRECATION_LOGS_PROFILE_ID,
+  profile: {
+    getDefaultAppState: () => () => ({
+      columns: [
+        { name: 'log.level' },
+        { name: 'message' },
+        { name: 'x-opaque-id' },
```

Comment:
Testing on an actual cluster, the actual field is `elasticsearch.http.request.x_opaque_id`. Apologies for the lack of proper requirements definition.